proposals/infra/INF-0004-register-types-and-diagnostics.md

```diff
@@ -59,12 +59,16 @@ specified resource class.
 | UAV | u | "Object type '%0' with resource class "UAV" expects register type 'u', but register type '%1' was given." |
 | CBuffer | b |  "Object type '%0' with resource class "CBuffer" expects register type 'b', but register type '%1' was given." |
 
-There are certain resource kinds within the SRV resource
-class that are "sampleable". These resources are typically
-prefixed with "Texture" in actual HLSL source. So, there will
-be a special check to see whether the candidate resource is 
-prefixed with "Texture", and if so, the "s" register type will
-also be permitted for that resource.
+There are certain exceptions to the table above. While we would like to
+reserve a register type for each resource class, and create a one-to-one
+mapping, there is legacy behavior that must be preserved in two specific
+cases. A Sampler resource class may use the 't' register type, and an
+SRV resource class may use the 's' register type. However, the compiler
+will not actively use such resources. This legacy behavior was permitted
+in prior versions of the compiler, but will now emit a warning that will
+be treated as an error by default. So, there will be a special exception
+for Samplers and SRVs, which may accept 't' and 's' respectively if the
+warning (--Wno-disallow-legacy-binding-rules) is disabled.

```

Comment:
We've discussed this, but I figured I'd get a note on here that I'm unconvinced we need to preserve this behavior.  FXC didn't support this, and the behavior in DXC seems more like a bug than a legacy feature.

Reply:
Yes, we decided that this behavior should no longer be supported, so this paragraph has now been removed from the spec.